docs/customizing.rst

```diff
@@ -0,0 +1,198 @@
+..
+    This file is part of Invenio.
+    Copyright (C) 2020 CERN.
+
+    Invenio is free software; you can redistribute it and/or modify it
+    under the terms of the MIT License; see LICENSE file for more details.
+
+Customizing
+===========
+
+
+how to inject/override the Invenio config (url, facets, the DOM id where to render the app): via parametrize or remove the Jinja macro and inject JSON config
+
+Configuring the settings of your Search Application
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A helper is provided to generate the configuration needed for the
+React-Searchkit application.
+This configuration contains the following information:
+
+`aggs`: a list of aggregations that will add accordingly facets to your search.
+
+.. code-block:: python
+
+    [
+        {
+            "aggName": "access_right",
+            "field": "access_right",
+            "title": "Access Right"
+        }
+    ]
+
+`appId`: the value here is needed since it will be used internally to
+differentiate the resources of multiple apps.
+"rdm-search"
+`initialQueryState`: This object is used to compose the initial search request
+that will be sent to the backend.
+
+.. code-block:: python
+
+    {
+        "hiddenParams": null,
+        "size": 10
+    }
+
+`layoutOptions`: here we can choose which layouts are going to be enabled
+in our search page, if both are present then a layout switcher will also be
+present in the UI.
```

Comment:
question: Is it automatic? IIRC you need to add the layourSwitcher component otherwise it just does not display. If so, it should be stated that the component *has to be added by the user*. (default templates have it though)

Reply:
ping

Reply:
It's automatic, this is done [here](https://github.com/inveniosoftware/invenio-search-ui/blob/44b6122d8f066a92344f1c1a58b4d18e888b690c/invenio_search_ui/assets/semantic-ui/js/invenio_search_ui/components/Results.js#L30)